source/s3_util.c

```diff
@@ -488,6 +488,86 @@ int aws_s3_parse_content_length_response_header(
     return result;
 }
 
+int aws_s3_parse_request_range_header(
+    struct aws_http_headers *request_headers,
+    bool *out_initial_message_has_start_range,
+    bool *out_initial_message_has_end_range,
+    uint64_t *out_initial_start_range,
+    uint64_t *out_initial_end_range) {
```

Comment:
utterly trivial: why bother naming these "initial_message"? it makes the tests weird. This is a standalone function that doesn't rely on state, but naming it "initial" makes the variables seem like we're informing this function about some previous state
```suggestion
    bool *out_has_start_range,
    bool *out_has_end_range,
    uint64_t *out_start_range,
    uint64_t *out_end_range) {
```

Reply:
Thanks, fixed.